lambdas/get_appointment_supporter.py

```diff
@@ -1,23 +1,22 @@
 import json
-import boto3
+from package.query_db import query
 
+#Written by Matt Hill
 #Input: supporter_id
 #Output: JSON object of current supporter appointments in the format: 
 # "supporterFN", "supporterLN", "studentFN", "studentLN", "type", "duration","method","location"
 
 def get_appointment_supporter(event, context):
 
-    given_id = event['supporter_id']
-    client = boto3.client('rds-data') #Connecting to the database
-    appointment_info = client.execute_statement(
-        secretArn = "arn:aws:secretsmanager:us-east-2:500514381816:secret:rds-db-credentials/cluster-33FXTTBJUA6VTIJBXQWHEGXQRE/postgres-3QyWu7",
-        database = "postgres",
-        resourceArn = "arn:aws:rds:us-east-2:500514381816:cluster:postgres",
-        sql = "SELECT U1.first_name as supporterFN, U1.last_name as supporterLN, U2.first_name as studentFN, U2.last_name as studentLN, SA.type, SA.duration, SA.method, SA.location \
-            FROM supporters S, users U1, users U2, student_appointment_relation SR, scheduled_appointments SA \
-             WHERE S.supporter_id = SR.supporter_id and SR.appointment_id = SA.appointment_id and S.supporter_id = U1.id and SR.student_id = U2.id and S.supporter_id = '%s';" % (given_id)
-    )
+    given_id = event['supporter_id'] 
+    sql = 'SELECT U1.first_name as supporterFN, U1.last_name as supporterLN, U2.first_name as studentFN, U2.last_name as studentLN, SA.type, SA.duration, SA.method, SA.location \
+          FROM supporters S, users U1, users U2, student_appointment_relation SR, scheduled_appointments SA \
+            WHERE S.supporter_id = SR.supporter_id and SR.appointment_id = SA.appointment_id and S.supporter_id = U1.id and SR.student_id = U2.id and S.supporter_id=:given_id;'
     
+    sql_parameters = [{'name':'given_id', 'value' : {'longValue': given_id}}]
+    appointment_info = query(sql, sql_parameters)
+    
+    #Check to see if the query even returned anything
     if (appointment_info['records'] == []): 
         return {
             'body': json.dumps("The supporter does not have any appointments or does not exist"),
```

Comment:
You should check that the supporter exists before executing the query. We should be able to tell the difference between a supporter not existing and a supporter not having appointments. This will make debugging and error handling much easier